test/auth.test.ts

```diff
@@ -0,0 +1,356 @@
+import Auth from '../src/auth';
+import Messenger from '../src/messenger';
+import { EmitterMessage, WorkerMessage } from '../src/types';
+
+describe('Auth', () => {
+  let windowOpenSpy: any;
+
+  beforeEach(() => {
+    windowOpenSpy = jest.spyOn(window, 'open');
+    windowOpenSpy.mockImplementation(() => {});
+    SharedWorker = jest.fn(
+      () =>
+        ({
+          port: {
+            start: () => {},
+            postMessage: () => {},
+          },
+        } as any)
+    );
+  });
+
+  afterEach(() => {
+    windowOpenSpy.mockRestore();
+    jest.clearAllMocks();
+  });
+
+  describe('constructor', () => {
+    it('sets the auth token if passed', () => {
+      const authToken = 'asdf';
+      const clientId = '1234';
+      const storageSpy = jest.spyOn(Storage.prototype, 'setItem');
+
+      new Auth({
+        clientId,
+        host: 'http://localhost:1738',
+        authToken,
+        messenger: new Messenger({}),
+      });
+      expect(storageSpy).toHaveBeenCalledWith('wally:1234:token', authToken);
+
+      storageSpy.mockRestore();
```

Comment:
for sure. is that generally the play - put stuff in `beforeEach`/`afterEach` whenever possible, even for a single test?

Reply:
I generally do that because it separates it and makes it more clear but do also understand that its somewhat subjective so I would say, it's a non-blocking comment for me :).